torch/_decomp/decompositions.py

```diff
@@ -1218,3 +1218,60 @@ def stack(tensors: List[Tensor], dim: int = 0) -> Tensor:
         return out.view(result_sizes)
     else:
         return torch.cat(get_stack_inputs(tensors, wrapped_dim), dim)
+
+
+@register_decomposition(aten.frac)
+def frac(input: Tensor) -> Tensor:
+    return input - torch.trunc(input)
+
+
+@register_decomposition(aten.celu)
+def celu(input: Tensor, alpha: float = 1.0) -> Tensor:
+    inv_alpha = 1.0 / alpha
+    # TODO: Should I use F.elu here?
+    return aten.elu(input, alpha, 1.0, inv_alpha)
+
+
+@register_decomposition(aten.mish)
+@pw_cast_for_opmath
+def mish(x: Tensor) -> Tensor:
+    # TODO: copied from cuda impl, note this is different from doc's 'x * x.softplus().tanh()
+    return x * x.exp().log1p().tanh()
+
+
+@register_decomposition(aten.softplus)
+def softplus(a: Tensor, beta: float = 1.0, threshold: float = 20.0) -> Tensor:
+    return torch.where((a * beta) > threshold, a, (a * beta).exp().log1p() / beta)
+
+
+@register_decomposition(aten.softshrink)
+def softshrink(a: Tensor, lambd: float = 0.5) -> Tensor:
+    # TODO: can these two where be commbined?
```

Comment:
possibly, but it would probably change the nan propagation behavior

Reply:
right... 
IIUC, we have NaN covered in op.sample_inputs()'s extremal test cases?